playlistmanager.lua

```diff
@@ -343,19 +356,34 @@ function escapepath(dir, escapechar)
   return string.gsub(dir, escapechar, '\\'..escapechar)
 end
 
+function replace_lefthand_matches(value, valid_values)
+  if value == nil or valid_values == nil then
+    return false
+  end
+  if valid_values['all'] then
+    return true
+  end
+  return valid_values['all'] or valid_values[value]
+end
+
+--escape special characters in url
+local filename_replace_functions = {
+  hex_to_char = function(x) return string.char(tonumber(x, 16)) end
+}
+
 --strip a filename based on its extension or protocol according to rules in settings
 function stripfilename(pathfile, media_title)
   if pathfile == nil then return '' end
   local ext = pathfile:match("%.([^%.]+)$")
-  local protocol = pathfile:match("^(%a%a+)://")
+  local protocol = is_protocol(pathfile)
```

Comment:
This should catch the protocol into a string instead of being a boolean

Reply:
You are right. Done.